app/controllers/smart_proxies_controller.rb

```diff
@@ -63,8 +66,12 @@ def destroy
 
   private
 
-  def requested_data(data_name)
-    data = @smart_proxy.public_send(data_name)
+  def find_status
+    @proxy_status = @smart_proxy.proxy_status
+  end
+
+  def requested_data(object, method)
+    data = object.public_send(method)
     render :json => {:success => true, :message => data}
```

Comment:
or even get rid of the hash and use HTTP response status to indicate success or failure?